### DISCUSSION TITLE
diesel sql log

### DISCUSSION THREAD

**Discussion by jutbin** [diesel-rs/diesel#4143] [Q&A]
How does diesel print the final executed sql, including additions, deletions, modifications and queries? Similar to printing the executed sql log in the console

**Comment by weiznich** [comment]
You might want to have a look at the `https://docs.diesel.rs/2.2.x/diesel/connection/fn.set_default_instrumentation.html` and `https://docs.diesel.rs/2.2.x/diesel/connection/trait.Connection.html#tymethod.set_instrumentation` functions. These allow you to configure a global or connection specific callback that is invoked for different connection events. You can use that for logging.

**Reply by jutbin** [reply]
Thank you for solving my problem. Have you used diesel_logger? How to use it in connection pool?

**Reply by weiznich** [reply]
For questions about `diesel_logger` please use their support channels. We cannot provide support for all the third party crates built on top of diesel here.

**Reply by jutbin** [reply]
How can I not print StartEstablishConnection and FinishEstablishConnection

**Reply by weiznich** [reply]
`InstrumentationEvent` is an ordinary rust enum, so you can just use constructs like [`match`](https://doc.rust-lang.org/book/ch06-02-match.html) to select only the values you want (or want to filter out).

**Reply by jutbin** [reply]
thanks

**Reply by jutbin** [reply]
How can I get the time taken to add in FinishQuery

**Reply by weiznich** [reply]
You would need to track that time on your own as part of your `Instrumentation` implementation.

**Reply by jutbin** [reply]
Could you please give me an example, this is very important to me

**Reply by weiznich** [reply]
I'm sorry, but I generally do not provide code to questions that didn't start with showing what they already have and that don't explain the problem they want to solve in detail. Or to word it differently: I prefer not to spend more effort on answering questions than people spend on writing them + thinking about a solution on their own before.

**Reply by jutbin** [reply]
In diesel, besides MysqlConnection, what other functions implement the Connection trait?

**Reply by weiznich** [reply]
https://docs.diesel.rs/2.2.x/diesel/connection/trait.Connection.html#implementors

**Reply by jutbin** [reply]
Can the printed log show which line of the diesel code the sql appears in?

**Reply by jutbin** [reply]
Can the printed SQL log show which file and which line executed the SQL?

**Reply by weiznich** [reply]
That's not supported

**Reply by jutbin** [reply]
I have a model field of NaiveDateTime type. The structure retrieved from the database does not include the time zone, that is, it does not include +07:00.

**Reply by jutbin** [reply]
Does diesel models not support returning serialized time with time zone?

**Reply by weiznich** [reply]
> I have a model field of NaiveDateTime type. The structure retrieved from the database does not include the time zone, that is, it does not include +07:00.

That's expected as [`NaiveDateTime`](https://docs.rs/chrono/0.4.38/chrono/struct.NaiveDateTime.html) is explicitly a type **without** timezone.

> Does diesel models not support returning serialized time with time zone?

That depends on whether your database supports timestamps with timezones or not.

**Reply by jutbin** [reply]
I am using mysql, I don't use NaiveDate what type should I use

**Reply by weiznich** [reply]
As far as I'm aware mysql don't support storing timezone information for your timestamps. 

(Please check the documentation for future questions like that one first. I do not have the capacity to answer them all, so I will likely stop responding here for such questions)

**Reply by jutbin** [reply]
I tried to use the time with time zone, but got an error
![Snipaste_2024-08-05_15-30-38](https://github.com/user-attachments/assets/16a82678-e6e6-4080-83f1-f39e5aff94f6)

**Reply by weiznich** [reply]
As written above: It's not supported by the database, so it's not supported by diesel.

(That's now my last answer here)
